groups.go

```diff
@@ -61,11 +61,22 @@ func (srv *groupsAPI) DeleteGroup(ctx context.Context, in *accountsv1.DeleteGrou
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	_, err = srv.authenticate(ctx)
+	token, err := srv.authenticate(ctx)
 	if err != nil {
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	accountId := token.UserID.String()
+
+	member, err := srv.memberRepo.Get(ctx, &models.MemberFilter{AccountID: &accountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	if member.Role != auth.RoleAdmin {
```

Comment:
Nawr! `auth.RoleAdmin` is different from being a group administrator! `auth.RoleAdmin` is a special token scope that identifies us (the developpers) whereas role within a group is something else (it's undefined for now, it's just a string)